src/platforms/common/server/shm_buffer.cpp

```diff
@@ -249,6 +291,16 @@ void mgc::MemoryBackedShmBuffer::bind()
         upload_to_texture(pixels.get(), stride_);
         uploaded = true;
     }
+    else if (is_dirty)
+    {
+        update_texture(pixels.get(), stride_);
```

Comment:
You actually could just use `upload_to_texture`, I think? `glTexImage2D` and `glTexSubImage2D` *should* be just as fast as each other for replacing a full texture with identical properties.